src/main/java/cloud/orbit/spring/OrbitSpringConfiguration.java

```diff
@@ -29,50 +29,112 @@
 @Configuration
 @EnableConfigurationProperties(OrbitActorsProperties.class)
 public class OrbitSpringConfiguration
 {
     @Autowired(required = false)
     private List<OrbitSpringConfigurationAddon> configAddons;
 
+    @Autowired(required = false)
+    @Qualifier("stageClock")
+    private Clock stageClock;
+
+    @Autowired(required = false)
+    @Qualifier("stageExecutorService")
+    private ExecutorService stageExecutorService;
+
+    @Autowired(required = false)
+    private Execution execution;
+
+    @Autowired(required = false)
+    private LocalObjectsCleaner localObjectsCleaner;
+
+    @Autowired(required = false)
+    private ClusterPeer clusterPeer;
+
+    @Autowired(required = false)
+    @Qualifier("executionObjectCloner")
+    private ExecutionObjectCloner executionObjectCloner;
+
+    @Autowired(required = false)
+    private MessageSerializer messageSerializer;
+
+    @Autowired(required = false)
+    @Qualifier("messageLoopbackObjectCloner")
+    private ExecutionObjectCloner messageLoopbackObjectCloner;
+
+    @Autowired(required = false)
+    private InvocationHandler invocationHandler;
+
+    @Autowired(required = false)
+    @Qualifier("stageTimer")
+    private Timer stageTimer;
+
     @Bean
-    public ActorExtension springActorConstructionExtension(AutowireCapableBeanFactory factory)
+    @ConditionalOnMissingBean(ActorConstructionExtension.class)
+    public ActorConstructionExtension springActorConstructionExtension(AutowireCapableBeanFactory factory)
     {
         return new SpringActorConstructionExtension(factory);
     }
 
     @Bean
-    public Stage stage(OrbitActorsProperties properties, List<ActorExtension> actorExtensions)
+    @ConditionalOnMissingBean(Messaging.class)
+    public Messaging messaging(OrbitActorsProperties properties)
     {
-        Stage.Builder stageBuilder = new Stage.Builder()
-                .extensions(actorExtensions.toArray(new ActorExtension[actorExtensions.size()]));
-
-        if (properties.getBasePackages() != null)
+        Messaging messaging = new Messaging();
+        if (properties.getMessagingTimeoutInMilliseconds() != null)
         {
-            stageBuilder.basePackages(properties.getBasePackages());
+            messaging.setResponseTimeoutMillis(properties.getMessagingTimeoutInMilliseconds());
         }
+        return messaging;
+    }
 
-        if (properties.getClusterName() != null)
-        {
-            stageBuilder.clusterName(properties.getClusterName());
-        }
+    @Bean
+    @ConditionalOnMissingBean(Stage.class)
+    public Stage stage(OrbitActorsProperties properties,
+                       List<ActorExtension> actorExtensions,
+                       Messaging messaging)
+    {
+        Stage.Builder stageBuilder = new Stage.Builder()
+                .extensions(actorExtensions.toArray(new ActorExtension[actorExtensions.size()]))
+                .clock(stageClock)
```

Comment:
Those can't be null because the Builder args are primitive types, not objects. It's kind of arbitrary...